app/models/concerns/password_resets_limitable.rb

```diff
@@ -0,0 +1,28 @@
+module PasswordResetsLimitable
+  extend ActiveSupport::Concern
+
+  included do
+    def password_resets_limit_reached?
+      self.resets_requested >= ArchiveConfig.PASSWORD_RESET_LIMIT && self.last_reset_within_cooldown?
+    end
+
+    def password_resets_available_time
+      self.reset_password_sent_at + ArchiveConfig.PASSWORD_RESET_COOLDOWN_HOURS.hours
+    end
+
+    def update_password_resets_requested
+      if self.resets_requested.positive? && !self.last_reset_within_cooldown?
+        self.resets_requested = 1
```

Comment:
Do you think it would make sense to do something like
```rb
self.update!(resets_requested: 1)
```
(and similarly for below as well)? That way, we won't lose the change in case someone forgets to call `save` later